File: qc-application/src/main/java/com/eager/questioncloud/application/message/FailQuestionPaymentMessage.java
package com.eager.questioncloud.application.message;

import com.eager.questioncloud.core.domain.payment.model.QuestionPayment;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class FailQuestionPaymentMessage {
    private QuestionPayment questionPayment;
    private int failCount;

    public static FailQuestionPaymentMessage create(QuestionPayment questionPayment) {
        return new FailQuestionPaymentMessage(questionPayment, 0);
    }

    public void increaseFailCount() {
        this.failCount++;
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/message/FailQuestionPaymentMessage.kt
package com.eager.questioncloud.application.message

import com.eager.questioncloud.core.domain.payment.model.QuestionPayment

class FailQuestionPaymentMessage(
    var questionPayment: QuestionPayment,
    var failCount: Int,
) {
    fun increaseFailCount() {
        failCount++
    }

    companion object {
        fun create(questionPayment: QuestionPayment): FailQuestionPaymentMessage {
            return FailQuestionPaymentMessage(questionPayment, 0)
        }
    }
}
